src/sinks/new_relic/encoding.rs

```diff
@@ -0,0 +1,56 @@
+use crate::{
+    sinks::util::encoding::{
+        Encoder, EncodingConfigFixed
+    }
+};
+use serde::{
+    Deserialize, Serialize
+};
+use std::{
+    fmt::Debug,
+    io
+};
+use super::NewRelicApiModel;
+
+#[derive(Deserialize, Serialize, Debug, Eq, PartialEq, Clone, Derivative)]
+#[serde(rename_all = "snake_case")]
+#[derivative(Default)]
+pub enum Encoding {
+    #[derivative(Default)]
+    Default,
+}
+
+impl Encoder<Result<NewRelicApiModel, &'static str>> for EncodingConfigFixed<Encoding> {
+    fn encode_input(&self, input: Result<NewRelicApiModel, &'static str>, writer: &mut dyn io::Write) -> io::Result<usize> {
+        if let Ok(api_model) = input {
+            let json = match api_model {
+                NewRelicApiModel::Events(ev_api_model) => to_json(&ev_api_model),
+                NewRelicApiModel::Metrics(met_api_model) => to_json(&met_api_model),
+                NewRelicApiModel::Logs(log_api_model) => to_json(&log_api_model),
+            };
+            if let Some(json) = json {
+                let size = writer.write(&json)?;
```

Comment:
`.write()` is not appropriate here. It attempts to write the data once, but makes no guarantees about how much of the data is actually written. You want `write_all()` instead. (You can read the difference between them here: https://doc.rust-lang.org/std/io/trait.Write.html)

However, `write_all` doesn't return the number of bytes written, which is required by the `Encoder` trait, so we have a helper function that does this for you.
`as_tracked_write` in `...sinks/util/encoding/codec`

Reply:
Sorry if the question sounds stupid, but, if `write_all` ensures that all bytes are written, the number of bytes isn't just the size of the byte array I'm writing? (in case it doesn't fail, of course)

Reply:
Resolved in https://github.com/vectordotdev/vector/pull/9764/commits/cf306a8c89459a6a2efdb4790fda2f95925697f6